scripts/migration/migrate_root_and_parent_on_node.py

```diff
@@ -0,0 +1,79 @@
+"""
+This will add a parent field and a parent_node field to all nodes.
+Root  will be the primary key of the originating parent node.
+Parent_node will be the first primary parent
+Done so that you can filter on both root nodes and parent nodes with a DB query
+
+"""
+
+import sys
+import logging
+from modularodm import Q
+from website import models
+from website.app import init_app
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def do_migration():
+    init_app(routes=False)
+    logger.warn('ultimate_parent field will be added to all nodes.')
+    all_undeleted_nodes = models.Node.find(Q('is_deleted', 'eq', False))
```

Comment:
Also, you can probably make this migration resume-able if you also query by `root_node`.

```python
Q('root_node', 'eq', None)
```

It's worth trying.